src/rlookup.c

```diff
@@ -988,69 +988,50 @@ int RLookup_LoadRuleFields(RedisModuleCtx *ctx, RLookup *it, RLookupRow *dst, In
+void RLookup_AddKeysFrom(const RLookup *src, RLookup *dest, uint32_t flags) {
+  RS_ASSERT(dest != NULL && src != NULL);
+  RS_ASSERT(dest != src);  // Prevent self-addition
+
+  // Iterate through all keys in source lookup
+  for (const RLookupKey *src_key = src->head; src_key; src_key = src_key->next) {
+    if (!src_key->name) {
+      // Skip overridden keys (they have name == NULL)
+      continue;
+    }
 
-    if (src->path) {
-      if (src->path == src->name) {
-        dst->path = dst->name;  // Point to our allocated name
-      } else {
-        dst->path = rm_strdup(src->path);
-      }
-    } //else - dst->path is already NULL
-  } else {
-    //source has no name
-    //dst->name = NULL;
-    //dst->path = NULL;
-    dst->flags = src->flags;  // Don't set NAMEALLOC if no strings
+    // Combine caller's control flags with source key's persistent properties
+    // Only preserve non-transient flags from source (F_SVSRC, F_HIDDEN, etc.)
+    // while respecting caller's control flags (F_OVERRIDE, F_FORCE_LOAD, etc.)
+    uint32_t combined_flags = flags | (src_key->flags & ~RLOOKUP_TRANSIENT_FLAGS);
+    RLookupKey *dest_key = RLookup_GetKey_Write(dest, src_key->name, combined_flags);
   }
-
-  return dst;
 }
 
-void RLookup_CloneInto(RLookup *dst, const RLookup *src) {
-  RS_ASSERT(dst != NULL && src != NULL);
-
-  // dst should be clean after RLookup_Init - only spcache should be set
-  // Assert that dst is in the expected clean state from initialization
-  RS_ASSERT(dst->head == NULL && dst->tail == NULL && dst->rowlen == 0 && dst->options == 0);
-
-  // Reset dst structure but preserve initialization
-  dst->rowlen = src->rowlen;
-  dst->options = src->options;
+void RLookupRow_WriteFieldsFrom(const RLookupRow *srcRow, const RLookup *srcLookup,
+                               RLookupRow *destRow, const RLookup *destLookup) {
+  RS_ASSERT(srcRow != NULL && srcLookup != NULL);
+  RS_ASSERT(destRow != NULL && destLookup != NULL);
 
-  // Deep copy the linked list of keys from src
-  RLookupKey *src_key = src->head;
-  RLookupKey *prev_dst_key = NULL;
-
-  while (src_key) {
-    RLookupKey *dst_key = RLookupKey_Clone(src_key);
+  // Iterate through all source keys
+  for (const RLookupKey *src_key = srcLookup->head; src_key; src_key = src_key->next) {
+    if (!src_key->name) {
+      // Skip overridden keys
+      continue;
+    }
 
-    if (!dst->head) {
-      // First key
-      dst->head = dst_key;
-      dst->tail = dst_key;
-    } else {
-      // Link to previous key
-      prev_dst_key->next = dst_key;
-      dst->tail = dst_key;
+    // Get value from source row
+    RSValue *value = RLookup_GetItem(src_key, srcRow);
+    if (!value) {
+      // No data for this key in source row
+      continue;
     }
 
-    prev_dst_key = dst_key;
-    src_key = src_key->next;
+    // Find corresponding key in destination lookup
+    RLookupKey *dest_key = RLookup_FindKey(destLookup, src_key->name, src_key->name_len);
```

Comment:
If needed - a potential performance improvement could be to support constant time lookup since now it is called here and also on all source fields from `RLookup_AddKeysFrom`